src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -159,6 +187,79 @@ export class ChallengesubmitComponent implements OnInit {
     this.challengeService.isChallengeHost.subscribe(status => {
       this.isChallengeHost = status;
     });
+
+    this.challengeService.isChallengeHost.subscribe(status => {
+      this.isChallengeHost = status;
+    });
+
+    if (this.challenge.is_docker_based) {
+      this.displayDockerSubmissionInstructions(this.challenge.id, this.isParticipated);
+    }
+    this.authToken = this.globalService.getAuthToken();
+  }
+
+  /**
+   * @param SELF current context
+   * @param eachPhase particular phase of a challenge
+   */
+  countDownTimer(SELF, eachPhase) {
+    const remainingTime = eachPhase.limits.remaining_time;
+    const days = Math.floor(remainingTime / 24 / 60 / 60);
+    const hoursLeft = Math.floor((remainingTime) - (days * 86400));
+    const hours = Math.floor(hoursLeft / 3600);
+    const minutesLeft = Math.floor((hoursLeft) - (hours * 3600));
+    const minutes = Math.floor(minutesLeft / 60);
+    let remainingSeconds = Math.floor(remainingTime % 60);
+    let remSeconds;
+    if (remainingSeconds < 10) {
+        remSeconds = '0' + remainingSeconds;
+    }
+    SELF.phaseRemainingSubmissionsCountdown[eachPhase.id] = {
+        'days': days,
+        'hours': hours,
+        'minutes': minutes,
+        'seconds': remSeconds
+    };
+    if (remainingTime === 0) {
+        SELF.phaseRemainingSubmissionsFlags[eachPhase.id] = 'showSubmissionDetails';
+    } else {
+        remainingSeconds--;
+    }
```

Comment:
`remainingSeconds--;` is doing nothing. (It's a local variable.)

Reply:
it will be set to the `seconds` variable of the dictionary

Reply:
Okay, it's there now.